src/documents/ResourceLoader.md

```diff
@@ -4,114 +4,261 @@ category: Infusion
-    <td>(Object) The resource loader instance.</td>
+    <td><code>path</code></td>
+    <td>A relative or absolute filesystem path to be loaded via the node.js
+    <a href="https://nodejs.org/api/fs.html"><code>fs</code></a> API.
+    As with <code>url</code>, this may include interpolated terms.</td>
+    <td>String</td>
+</tr>
+<tr>
+    <td><code>resourceText</code></td>
+    <td>The complete literal resource text to be reported as loaded. This is useful in an infrastructural context
+    where the required resource
+    has already been loaded or computed via some other means, and one merely needs to inject it into the
+    <code>resourceLoader</code> workflow.</td>
+    <td>String</td>
+</tr>
+<tr>
+    <td><code>promiseFunc</code></td>
+    <td>A (IoC reference to a) function returning a promise which yields the required resource text. This will be
+    invoked with the arguments resolved from the <code>resourceSpec</code> field <code>promiseArgs</code> which may
+    also be present.
+    </td>
+    <td>Function</td>
+</tr>
+<tr>
+    <td><code>dataSource</code></td>
+    <td>A reference to a <code>fluid.dataSource</code> component whose <code>get</code> method will be invoked with
+    the contents
+    of the <code>directModel</code> field in this <code>resourceSpec</code>. Documentation for core Infusion
+    DataSources will
+    be ported soon, in the meantime you may refer to the docs on
+    <a href="https://github.com/fluid-project/kettle/blob/master/docs/DataSources.md">
+    Kettle DataSources</a> and substitute <code>kettle</code> for <code>fluid</code> in all gradeNames.
+    </td>
+    <td>fluid.dataSource</td>
 </tr>
 </table>
 
-### Options
+Two of the duck typing fields above (`url`, `path`) are "path fields" and can contain interpolated terms, as well as
+participating in the resource localisation algorithm based on path suffix. The other three (`resourceText`,
+`promiseFunc` and `dataSource`) are "non-path fields" and do not participate in interpolation or localisation.
 
 <table>
-<tr><th>Name</th><th>Description</th><th>Values</th><th>Default</th></tr>
+    <tr>
+        <th colspan="3">Members within a <code>resourceSpec</code> structure nested within the <code>resources</code>
+        structure of a <code>fluid.resourceLoader</code> component</th>
+    </tr>
+<tr><th>Name</th><th>Description</th><th>Type</th></tr>
 <tr>
-    <td><code>terms</code></td>
-    <td>
-        Contains a map of variable names and replacement values to compose the path to each physical file. The map
-        structure complies with the format of the second <code>terms</code> argument of <a
-        href="CoreAPI.html#fluidstringtemplatetemplate-terms">fluid.stringTemplate()</a> API. Also see <a
-        href="./tutorial-usingStringTemplates/UsingStringTemplates.html">Using String Templates</a> for more
-        information.
-    </td>
-    <td>Object</td>
-    <td><code>{}</code></td>
+    <td><code>[pathField]</code></td>
```

Comment:
Are the `[]` intentional?

Reply:
Represents that this is not an actual member name in the structure, but is a placeholder for a "path-type" duck typing field like `url` `path`, or a "non-path" field. I've added some explanatory text above the table to clarify this.